api/v1alpha1/tcproute_types.go

```diff
@@ -16,44 +16,51 @@ limitations under the License.
 package v1alpha1
 
 import (
+	core "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
-// TcpRouteSpec defines the desired state of TcpRoute
-type TcpRouteSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+// TCPRouteSpec defines the desired state of the TCPRoute.
```

Comment:
Is this TCPRoute or TLSRoute?

TCP means plain TCP
TLSRoute seems to be more apt?

Reply:
I don't have a preference on naming but we want to support both of these use-cases, TCP and TLS routing using the same CRD.
My leaning is slightly more towards TCPRoute.

Reply:
Might be confusing for users, we will have to make sure it's unambiguous what happens, also can be kind of weird to mix TCP and TLS together (do users have this as a use case)?

Traffic to port 80 goes?

```
gateway:
  port 80, protocol HTTP 
  port 443, protocol HTTPS
routes:
  - myRoute

myRoute:
TCPRoute
  "foo.com" => svc1
```

Reply:
Something more like:

TCP (port based routing):
```
- port: 4242 => svc1
- port: 2424 => svc2
```
This is the dumbest form of routing one can have essentially.
https://github.com/kubernetes-sigs/service-apis/issues/96


TLS (SNI based routing):
```
- host/sni: foo.com => svc1
- host/sni: bar.com => svc2
```
This can optionally include port too. User story: I want TLS sessions for foo.com on port X to be routed to svc1 but foo.com on port Y should not be allowed.